### PR TITLE
[FIX]fix is_ip_address_in_range func with const param

### PR DESCRIPTION
**Pull request by sjyango**
## Proposed changes

Issue Number: close #xxx

pr https://github.com/apache/doris/pull/32580 and https://github.com/apache/doris/pull/33968 legacy work.
- before this pr:
```shell
mysql> select id, is_ip_address_in_range(addr, '192.168.100.0/24') from test_is_ip_address_in_range_function order by id;
ERROR 1105 (HY000): errCode = 2, detailMessage = (10.214.224.189)[CANCELLED][INVALID_ARGUMENT][E33] Illegal column Const(String) of argument of function is_ip_address_in_range, expected String

        0#  doris::get_stack_trace(int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) at /home/ysj/doris/be/src/util/stack_util.cpp:54
        1#  doris::Exception::Exception(int, std::basic_string_view<char, std::char_traits<char> > const&) at /home/ysj/doris/be/src/common/exception.cpp:28
        2#  doris::Exception::E
```
- after this pr:
Execute normally without any errors when input is const param.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by sjyango** [comment]
run buildall

File: regression-test/suites/query_p0/sql_functions/ip_functions/test_is_ip_address_in_range_function.groovy
Comment (zclllyybb) [inline regression-test/suites/query_p0/sql_functions/ip_functions/test_is_ip_address_in_range_function.groovy:1]: don't add regression for this but be-ut

**Comment by sjyango** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by zhangstar333 — APPROVED** [review]

**Pull request merged**